src/t_zset.c

```diff
@@ -2585,6 +2553,22 @@ static void zdiff(zsetopsrc *src, long setnum, zset *dstzset, size_t *maxelelen,
     }
 }
 
+struct sdsDoublePair {
+    sds key;
+    double score;
+};
+
+const void *sdsDoubleMapGetKey(const void *entry) {
+    struct sdsDoublePair *keyval = (struct sdsDoublePair *)entry;
```

Comment:
Does this work with an implicit cast? Maybe add `const` to the struct to avoid warning when casting from const to non-const?

```suggestion
    const struct sdsDoublePair *keyval = entry;
```